src/filtering/update.jl

```diff
@@ -67,7 +67,11 @@ function update!(
     D = length(m_p)
 
     # K = P_p * H' / S
-    S_chol = cholesky!(S)
+    if S isa SRMatrix
```

Comment:
This here is likely to reduce performance a bit right now (in-place vs out-of-place)